### DISCUSSION TITLE
IdentityCore Customizado

### DISCUSSION THREAD

**Discussion by Alekseidasilva** [balta-io/forum#592] [Tira Dúvidas] [answered]
Boa noite @andrebaltieri !
Desculpe,  não sei se aqui é ligar certo para colocar esta questão,   mas preciso de uma explicação sobre Identity!
@andrebaltieri , estou querendo criar uma app que conterá 3 classes (employee, Student e Teacher) preciso fazer com que as 3 implementam o identityUser  para que as mesmas possam usufruir dos recursos do identity!
Sendo que no CursoFullStack, vc coloca a classe user no projecto API para não poluir o CORE com EF. 
O que posso fazer?
Por outra poderia me dar uma dicas de como customizar o identity completo? E que da forma que esta no curso, sinto que nao tenho controle do mesmo sendo que nao consigo debugar em alguns pontos....

PF,  me ajude aí, @andrebaltieri !

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @Alekseidasilva, como vai?

Claro, vamos lá...

Primeiro passo: Fazer o curso Fullstack, lá eu mostro a base do Identity
Segundo passo: A customização do Identity eu mostrei aqui: https://balta.io/agenda/jornada-fullstack-imersao-1
Inclusive aí eu deixei uma "receita de bolo" para mapeamento e tudo mais.

Em relação ao seu desafio, você não precisa customizar o Identity... uma coisa é o usuário, outra coisa é o Student, Employee e Teacher... esses três últimos vivem sem o usuário.

Em resumo, você não precisa customizar o Identity, só ter uma referência para o usuário nas classes que quiser, assim como mostro no curso Fullstack! 🚀
